File: cetaphil-event/src/main/java/com/cetaphilevent/util/EventServiceException.java
package com.cetaphilevent.util;

import lombok.Data;

@Data
public class EventServiceException extends Exception{
	private String code;
	private Object details;
		
	public EventServiceException(String code,String message,Object details){
		super(message);
		
		this.code = code;
		this.details = details;
	}
}

File: Samsonite.Mall/Samsonite.Mall.Domain/Exceptions/EventServiceException.cs
using System;

namespace Samsonite.Mall.Domain.Exceptions {
    public class EventServiceException : Exception {
        public string Code { get; set; }
        public new object Data { get; set; }

        public EventServiceException(string code, string message, object data)
            : base(message) {

            Code = code;
            Data = data;
        }

        public EventServiceException(string message, object data) : base(message) {
            Data = data;
        }
    }
}
